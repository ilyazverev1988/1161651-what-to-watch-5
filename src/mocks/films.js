export default [
  {
    preview: `img/aviator.jpg`,
    nameFilm: `Aviator`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Thrillers`,
    releaseYear: 2014,
    descriptionFilm: `Friendship, metamorphosis, and adventure.`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Everything we do is connected with zen: sex, bliss, moonlight, sorrow.`,
    producer: `Heu, orexis!`,
    duration: `1h 39m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/dardjeeling-limited.jpg`,
    nameFilm: `Dardjeeling limited`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2016,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/bohemian-rhapsody.jpg`,
    nameFilm: `Bohemian rhapsody`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 40m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    nameFilm: `Fantastic beasts the crimes of grindelwald`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2018,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 8.9,
    numberOfVotes: 3,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 50m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/johnny-english.jpg`,
    nameFilm: `Johnny english`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2019,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 9.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 30m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/macbeth.jpg`,
    nameFilm: `Macbeth`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2020,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 5.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 33m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/no-country-for-old-men.jpg`,
    nameFilm: `No country for old men`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2017,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 7.9,
    numberOfVotes: 2,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 44m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
  {
    preview: `img/war-of-the-worlds.jpg`,
    nameFilm: `War of the worlds`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    filmCover: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Sci-Fi`,
    releaseYear: 2015,
    descriptionFilm: `Spaces malfunction with adventure!`,
    commonScore: 6.9,
    numberOfVotes: 1,
    cast: `Never ransack a woodchuck.`,
    producer: `When the creator of zen acquires the stigmas of the aspect, the extend will know seeker.`,
    duration: `1h 54m`,
    linkPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.web`,
    linkFullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `
  },
];
