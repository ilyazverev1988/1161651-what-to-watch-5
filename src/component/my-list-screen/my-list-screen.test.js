import React from "react";
import renderer from "react-test-renderer";
import MyList from "./my-list-screen";
import films from "../../mocks/films";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";

const noop = () => {};
const mockStore = configureMockStore();
const store = mockStore({
  USER: {
    userData: {
      id: 1,
      email: `3@mail.ru`,
      name: `3`,
      avatarURL: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
      errorAuthorization: ``
    },
    authorizationStatus: `AUTH`
  }
});

describe(`Should MyList render correctly`, () => {
  it(`With filmActive`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <MyList
                filmActive={``}
                films={films}
                handleMouseEnterFilm={noop}
                handleMouseOverFilm={noop}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`No filmActive`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <MyList
                filmActive={1}
                films={films}
                handleMouseEnterFilm={noop}
                handleMouseOverFilm={noop}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
