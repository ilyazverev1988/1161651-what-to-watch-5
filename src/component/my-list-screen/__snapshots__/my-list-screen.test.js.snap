// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly No filmActive 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <a
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <video
              height="175"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Aviator1
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should MyList render correctly With filmActive 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <a
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <video
              height="175"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Aviator1
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
